classes/helper.php

```diff
@@ -0,0 +1,127 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace mod_hvp;
+
+/**
+ * Helper class for plugin.
+ *
+ * @package    mod_hvp
+ * @author     Andrew Madden <andrewmadden@catalyst-au.net>
+ * @copyright  2022 Catalyst IT
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class helper {
+
+    /**
+     * Get an installed library.
+     *
+     * @param array $options Array containing either id or a title and version number.
+     * @return \stdClass Record from hvp_libraries.
+     */
+    public static function get_library(array $options): \stdClass {
```

Comment:
Id like to see options become a little clearer. Can you start with a default options array, then foreach it and apply the supplied value if its present? Its very hard to track what makes a valid options array atm.

Reply:
Good idea

Reply:
Done